lock/src/test/java/org/infinispan/lock/api/ClusteredLockSplitBrainTest.java

```diff
@@ -0,0 +1,140 @@
+package org.infinispan.lock.api;
+
+import static org.infinispan.functional.FunctionalTestUtils.await;
+import static org.infinispan.test.Exceptions.assertException;
+import static org.infinispan.test.Exceptions.expectException;
+import static org.testng.AssertJUnit.assertFalse;
+import static org.testng.AssertJUnit.assertNotNull;
+import static org.testng.AssertJUnit.assertTrue;
+import static org.testng.AssertJUnit.fail;
+
+import java.util.concurrent.CompletionException;
+
+import org.infinispan.lock.exception.ClusteredLockException;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.partitionhandling.AvailabilityException;
+import org.infinispan.partitionhandling.BasePartitionHandlingTest;
+import org.testng.annotations.Test;
+
+@Test(groups = "functional", testName = "clusteredLock.ClusteredLockSplitBrainTest")
+public class ClusteredLockSplitBrainTest extends BasePartitionHandlingTest {
```

Comment:
there are missing tests with splitting the cluster with lock acquired. 
also, it would be nice to test the `tryLock()`s

Reply:
I didn't want to go to more tests without a check on this one first.